src/lib/util/Music/ManagerWrapper.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ *   Copyright 2019-2020 Jacz
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+import type { SkyraClient } from '@lib/SkyraClient';
+import { WSGuildCreate } from '@lib/types/DiscordAPI';
+import { DiscordPacket, LavalinkNodeOptions, Manager as LavacordManager, ManagerOptions } from 'lavacord';
+
+export class Manager extends LavacordManager {
+
+	public constructor(public readonly client: SkyraClient, nodes: LavalinkNodeOptions[], options?: ManagerOptions) {
+		super(nodes, options || {});
+		this.client = client;
+
+		this.send = packet => {
+			if (this.client.guilds.size) {
+				const guild = this.client.guilds.get(packet.d.guild_id);
+				if (guild) return guild.shard.send(packet);
+			} else {
+				const guild = this.client.guilds.get(packet.d.guild_id);
+				if (guild) return guild.shard.send(packet);
+			}
+		};
+
+		client.once('ready', () => {
+			this.user = client.user!.id;
+			this.shards = client.options.shardCount || 1;
+		});
+
+		if (client.guilds.size) {
```

Comment:
I'd recommend getting rid of this stuff as I believe you do it yourself in raw events.

Reply:
We don't. I removed most of the raw event processing. The only raw events we still process are `TrackStuckEvent` and `WebSocketClosedEvent`. Other than that we check for the `close` event and the `ready` event emitted by lavalink and `playerUpdate`, `start`, `error` and `end` emitted by the player.

Reply:
https://github.com/skyra-project/skyra/blob/f77df41f3c5219c1ecaa57f910c64ea6930d8be4/src/events/rawVoiceServerUpdate.ts

https://github.com/skyra-project/skyra/blob/f77df41f3c5219c1ecaa57f910c64ea6930d8be4/src/events/rawVoiceStateUpdate.ts